matrixprofile/matrixProfile.py

```diff
@@ -83,26 +137,29 @@ def _stamp_parallel(tsA, m, tsB=None, sampling=0.2, n_threads=-1, random_state=N
     """
     if n_threads is -1:
         n_threads = multiprocessing.cpu_count()
-    
+
     n = len(tsA)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
 
```

Comment:
All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB.

```python
if not is_array_like(tsB):
    tsB = tsA
```